internal/pkg/service/buffer/statistics/apinode.go

```diff
@@ -49,13 +44,12 @@ type dependencies interface {
 
 func NewAPINode(d dependencies) *APINode {
 	m := &APINode{
-		nodeID: d.Process().UniqueID(),
-		logger: d.Logger().AddPrefix("[stats]"),
-		clock:  d.Clock(),
-		store:  d.Store(),
-		// channel needs to be large enough to not block under average load
-		ch:       make(chan notifyEvent, 2048),
```

Comment:
Zmeny statistik sa zapisovali do tohto channel/buffer a potom ich spracovavala goroutine.

Upravil som to aby sa pouzil zamok a lokalne statistiky v memory sa upravia hned.
Je to tak ovela jednoduchsie testovatelne + nerobi to v testoch data race.

Lebo doteraz, ked sa ulozili `records`, tak si nemal test ako prinutit, aby hned updatol aj statistiky.